core/src/com/fourinachamber/fortyfive/game/GameDirector.kt

```diff
@@ -46,98 +49,74 @@ class GameDirector(private val controller: GameController) {
             enemyActionTimes = listOf((turns / 2.0).toInt())
         }
         FortyFiveLogger.debug(logTag, "turnRevealTime = $turnRevealTime; enemyActionTimes = $enemyActionTimes")
-        val enemy = scaleAndCreateEnemy(chosen, difficulty)
+        enemy = scaleAndCreateEnemy(enemyProto, difficulty)
         FortyFiveLogger.debug(logTag, "enemy: health = ${enemy.health}; damage = ${enemy.damage}")
         controller.initEnemyArea(enemy)
     }
 
-    fun checkActions() {
+    fun checkActions(): Timeline {
         if (controller.turnCounter == turnRevealTime) {
-            val remainingTurns = turns - controller.turnCounter
-            controller.remainingTurns = remainingTurns
-            controller.executeTimeline(Timeline.timeline {
+            return Timeline.timeline {
+                val remainingTurns = turns - controller.turnCounter
+                action {
+                    controller.remainingTurns = remainingTurns
+                }
                 include(controller.confirmationPopup("You have $remainingTurns left!"))
-            })
+            }
         } else if (controller.turnCounter in enemyActionTimes) {
-            doEnemyAction()
+            return doEnemyAction()
         }
+        return Timeline(mutableListOf())
     }
 
-    private fun doEnemyAction() {
+    fun end() {
+        val newDifficulty = adjustDifficulty()
+        FortyFiveLogger.debug(logTag, "adjusted difficulty from $difficulty to $newDifficulty")
+        SaveState.currentDifficulty = newDifficulty
+    }
+
+    private fun adjustDifficulty(): Double {
+        if (controller.playerLost) return difficulty // Too late to adjust difficulty of enemy lol
```

Comment:
sollte es nicht leichter werden, wenn man verliert, damit es beim nächsten mal nicht schlimmer wird?
(falls das nur für den aktuellen Kampf zählt, ist es natürlich egal)
(ich weiß, was im kommentar steht, ABER du liest in der init ja auch den savestate aus, deshalb unsicher)

Reply:
In the demo the player would loose the run immediately after loosing to the enemy, so it didn't matter here. For the new gameplay rework the scaling system will need to be completely changed anyway.